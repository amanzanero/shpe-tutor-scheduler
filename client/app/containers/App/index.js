/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
import { Switch, Route, BrowserRouter } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import red from '@material-ui/core/colors/red';

import HomePage from 'containers/HomePage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import UserNav from 'components/UserNav';

import GlobalStyle from '../../global-styles';

/**
 * Theme to wrap entire application
 * hex is from official USC website
 * https://identity.usc.edu/print/colors/
 */
const theme = createMuiTheme({
  palette: {
    primary: { main: '#990000' },
    secondary: { main: '#FFCC00' },
    error: red,
    contrastThreshold: 3,
    tonalOffset: 0.2,
  },
  typography: {
    useNextVariants: true,
  },
});

export default function App() {
  return (
    <div>
      <MuiThemeProvider theme={theme}>
        <BrowserRouter>
          <Switch>
            <UserNav>
              <Route exact path="/" component={HomePage} />
              <Route component={NotFoundPage} />
            </UserNav>
          </Switch>
        </BrowserRouter>
        <GlobalStyle />
      </MuiThemeProvider>
    </div>
  );
}
