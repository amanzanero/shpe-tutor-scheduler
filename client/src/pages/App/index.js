import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/styles';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';

import initStore from '../../config/store';

import HomePage from '../HomePage';
import LandingPage from '../LandingPage';
import theme from '../../theme';

const styles = {
  root: {
    height: '100vh',
    display: 'flex',
    flexFlow: 'column',
  },
};

const App = classes => (
  <div className={classes.root}>
    <ThemeProvider theme={theme}>
      <Router>
        <Route path="/" component={LandingPage} />
        <Route path="/home" component={HomePage} />
      </Router>
    </ThemeProvider>
  </div>
);

const AppWrapper = props => {
  const { classes } = props;
  const { store, persistor } = initStore();
  return (
    <Provider store={store}>
      {persistor ? (
        <PersistGate loading={null} persistor={persistor}>
          {App(classes)}
        </PersistGate>
      ) : (
        App(classes)
      )}
    </Provider>
  );
};

AppWrapper.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(AppWrapper);
