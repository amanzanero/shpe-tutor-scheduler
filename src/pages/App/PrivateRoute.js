import React from 'react';
import { Redirect, Route } from 'react-router-dom';

import { getSession } from '../../utils/Authenticator';

export default function PrivateRoute({ component: Component, ...rest }) {
  return (
    <Route
      {...rest}
      render={props =>
        getSession() ? <Component {...props} /> : <Redirect to="/" />
      }
    />
  );
}
