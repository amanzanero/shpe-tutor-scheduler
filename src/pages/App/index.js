import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/styles';
import { makeStyles } from '@material-ui/core/styles';
import { Provider } from 'react-redux';

import initStore from '../../config/store';

import HomePage from '../../containers/HomePage';
import LandingPage from '../LandingPage';
import theme from '../../theme';
import ManageAppointments from '../ManageAppointments';
import Nav from '../../containers/Nav';
import ManageCourses from '../../containers/ManageCourses';
import Settings from '../../containers/Settings';

const useStyles = makeStyles({
  root: {
    height: '100vh',
    display: 'flex',
    flexFlow: 'column',
  },
});

const App = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <ThemeProvider theme={theme}>
        <ManageCourses />
        <Settings />
        <Router>
          <Switch>
            <Route exact path="/" component={LandingPage} />
            <Route path="/" component={Nav} />
          </Switch>
          <Route path="/home" component={HomePage} />
          <Route path="/appointments" component={ManageAppointments} />
        </Router>
      </ThemeProvider>
    </div>
  );
};

export default function AppWrapper() {
  const store = initStore();
  return (
    <Provider store={store}>
      <App />
    </Provider>
  );
}
