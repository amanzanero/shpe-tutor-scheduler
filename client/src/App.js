import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <p>Hey there</p>
      </div>
    );
  }
}

export default App;
