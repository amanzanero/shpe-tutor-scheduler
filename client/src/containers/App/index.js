import React, { Component } from 'react';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { theme } from './theme';
import { BrowserRouter as Router, Route } from "react-router-dom";

import './App.css';
import NavBar from '../../components/NavBar';
import HomePage from '../HomePage';
import SignUp from '../SignUp';

class App extends Component {
  render() {
    return (
      <div className="App">
        <MuiThemeProvider theme={theme}>
          <NavBar/>
          <Router>
            <Route exact path="/" component={HomePage} />
            <Route path="/sign-up" component={SignUp} />
          </Router>
        </MuiThemeProvider>
      </div>
    );
  }
}

export default App;
