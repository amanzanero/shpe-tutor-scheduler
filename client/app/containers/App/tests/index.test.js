import React from 'react';
import { render } from 'react-testing-library';

import App from '../index';

describe('<App />', () => {
  it('renders and matches the snapshot', () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
