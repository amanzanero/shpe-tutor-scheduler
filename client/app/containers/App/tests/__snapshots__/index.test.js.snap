// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders and matches the snapshot 1`] = `
<div>
  <div
    class="UserNav-root-1"
  >
    <header
      class="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-11"
    >
      <div
        class="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41"
      >
        <button
          aria-label="Menu"
          class="MuiButtonBase-root-50 MuiIconButton-root-44 MuiIconButton-colorSecondary-47 UserNav-menuButton-3"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-49"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-53"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root-138"
          />
        </button>
        <h6
          class="MuiTypography-root-62 MuiTypography-h6-79 MuiTypography-colorSecondary-93 UserNav-grow-2"
        >
          SHPE Tutoring and Academics
        </h6>
        <button
          class="MuiButtonBase-root-50 MuiButton-root-98 MuiButton-text-100 MuiButton-textSecondary-102 MuiButton-flat-103 MuiButton-flatSecondary-105"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-99"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root-138"
          />
        </button>
      </div>
    </header>
  </div>
</div>
`;
